beacon_node/beacon_chain/src/kzg_utils.rs

```diff
@@ -1,76 +1,88 @@
-use kzg::{Error as KzgError, Kzg, BYTES_PER_BLOB};
+use kzg::{Error as KzgError, Kzg, KzgPreset};
 use types::{Blob, EthSpec, Hash256, KzgCommitment, KzgProof};
 
 /// Converts a blob ssz List object to an array to be used with the kzg
 /// crypto library.
-fn ssz_blob_to_crypto_blob<T: EthSpec>(blob: Blob<T>) -> kzg::Blob {
-    let blob_vec: Vec<u8> = blob.into();
-    let mut arr = [0; BYTES_PER_BLOB];
-    arr.copy_from_slice(&blob_vec);
-    arr.into()
+fn ssz_blob_to_crypto_blob<T: EthSpec>(
+    blob: Blob<T>,
+) -> Result<<<T as EthSpec>::Kzg as KzgPreset>::Blob, KzgError> {
+    T::blob_from_bytes(blob.to_vec().as_slice())
 }
 
 /// Validate a single blob-commitment-proof triplet from a `BlobSidecar`.
 pub fn validate_blob<T: EthSpec>(
-    kzg: &Kzg,
+    kzg: &Kzg<T::Kzg>,
     blob: Blob<T>,
     kzg_commitment: KzgCommitment,
     kzg_proof: KzgProof,
 ) -> Result<bool, KzgError> {
     kzg.verify_blob_kzg_proof(
-        ssz_blob_to_crypto_blob::<T>(blob),
+        ssz_blob_to_crypto_blob::<T>(blob)?,
         kzg_commitment,
         kzg_proof,
     )
 }
 
 /// Validate a batch of blob-commitment-proof triplets from multiple `BlobSidecars`.
 pub fn validate_blobs<T: EthSpec>(
-    kzg: &Kzg,
+    kzg: &Kzg<T::Kzg>,
     expected_kzg_commitments: &[KzgCommitment],
     blobs: &[Blob<T>],
     kzg_proofs: &[KzgProof],
 ) -> Result<bool, KzgError> {
+    // TODO(sean) batch verification fails with a single element, it's unclear to me why
```

Comment:
When I enabled blob building in the execution block generator, we had a bunch of tests producing random numbers of blobs (which has been good for testing) but i noticed a few tests would fail during kzg proof verification whenver the random number of blobs was of size 1. I only observed this on the minimal spec, didn't check with mainnet. I haven't investigated further, opened an issue here: https://github.com/sigp/lighthouse/issues/4446

If we can pin it down to the bindings or the the c-kzg lib generally we can open an issue there, just haven't had time to look into it.

Reply:
sounds good, i'll try to chase this up